packages/vite/src/node/build.ts

```diff
@@ -640,3 +643,12 @@ function wrapSsrHook(fn: Function | undefined) {
     return fn.call(this, ...args, true)
   }
 }
+
+function getNpmPackageName(importPath: string) {
+  const parts = importPath.split('/')
+  if (parts[0].startsWith('@')) {
+    return `${parts[0]}/${parts[1]}`
```

Comment:
`parts[1]` could be undefined.

Reply:
> parts[1] could be undefined.

I'm assuming the keys of `Object.keys(require('node_modules/.vite/_metadata.json').optimized` to always be modules of npm packages since it doesn't really make sense (AFAICT) for user-land code to be included in `optimized`. Am I wrong?

If my assumption is correct then `parts[1]` cannot be `undefined` since npm packages that start with `@` are scoped packages which always includes a `/`.

> This is really good to have, but I'd think it's better to have some tests to make this more reliable.

Maybe we can use `assert`s. E.g.

```js
export function resolveSSRExternal(
  config: ResolvedConfig,
  knownImports: string[],
  ssrExternals: Set<string> = new Set()
): string[] {
   // `knownImports` should always be npm package names
   assert(knownImports.every(id => getNpmPackageName(id)===id))

   // ...
```

More importantly:

```js
function getNpmPackageName(importPath: string) {
  assert(isNpmPackageName(importPath))

  //...
}

function isNpmPackageName(importPath: string): boolean {
   if( importPath.startsWith('.') ) {
     return false
   }

   // ....
}
```

Objections?

Reply:
Published a new version which doesn't have that problem anymore.

Reply:
Published a new version which doesn't have that problem anymore.